internal/pkg/cli/progress.go

```diff
@@ -12,3 +20,107 @@ type progress interface {
 	// Events writes additional information in between the start and stop stages.
 	Events([]string)
 }
+
```

Comment:
I'd like to get feedback on whether y'all think there is a more appropriate package to put these types and functions?

I thought maybe the `term/spinner` pkg since it's about transforming `[]deploy.ResourceEvent` to a `[]string` that the `term/spinner` pkg understands.